pkg/gadget-service/api-helpers/apihelpers.go

```diff
@@ -78,3 +83,54 @@ func Validate(p api.Params, v api.ParamValues) error {
 	}
 	return nil
 }
+
+// GetStringValuesPerDataSource will separate a string and extract per-datasource values. It expects a string like
+// `datasource1:value,datasource2:value` or `value` (datasource is optional - this will lead to an empty key)
```

Comment:
I'm wondering about this syntax and how we can improve it. Currently we have:

```
--flagname1=datasource1:foo,datasource2:bar,...
--flagname2=datasource1:foo2,datasource2:bar2,...
```

I'm wondering if we can change it to something like

```
--datasource1.flagname1=foo --datasource2.flagname1=bar
--datsource2.flagname2=foo2 --datasource2.flagname2=bar2
```

If there's a single data source, we can create an alias 

```
--datasource1.flagname1 == --flagname1
```

I feel having a single flag with the `:`, `,` syntax is more complicated than having multiple flags. I also think multiple flags can play better with the configuration file:

```yaml
datasource1: 
  flagname1: foo
  flagname2: bar 
datasource2:
  flagname1: foo2 
  flagname2: bar2
```

WDYT?

Reply:
I totally agree - but it could create LOTS of parameters that way. Maybe then it's better to go the annotation route and use something like `--annotate datasource1.flagname1=value`. I'll think more about it.